packages/core-components/src/layout/ProxiedSignInPage/ProxiedSignInIdentity.test.ts

```diff
@@ -164,5 +163,57 @@ describe('ProxiedSignInIdentity', () => {
       await identity.getSessionAsync(); // now the expiry has passed
       expect(serverCalled).toHaveBeenCalledTimes(2);
     });
+
+    it('handles optional headers correctly', async () => {
+      let req1: Request;
+      const getBaseUrl = jest.fn();
+      const serverCalled = jest.fn().mockImplementation(req => {
+        req1 = req;
+      });
+      const getHeaders = jest.fn().mockResolvedValue({ 'x-foo': 'bars' });
+
+      worker.events.on('request:match', serverCalled);
+      worker.use(
+        rest.get('http://example.com/api/auth/foo/refresh', (_, res, ctx) =>
+          res(
+            ctx.status(200),
+            ctx.set('Content-Type', 'application/json'),
+            // dummy response as we are only testing the request in this test
+            ctx.json({
+              providerInfo: {},
+              profile: {},
+              backstageIdentity: {
+                token: '',
+                identity: {
+                  ownershipEntityRefs: [''],
+                  userEntityRef: '',
+                  type: 'user',
+                },
+              },
+            }),
+          ),
+        ),
+      );
+
+      const identity = new ProxiedSignInIdentity({
+        provider: 'foo',
+        discoveryApi: { getBaseUrl },
+        getHeaders: getHeaders,
```

Comment:
Nit: I had to scroll up and use the browser find feature to see where getHeaders is declared 😅. Maybe we can declare it closer to this, maybe just above `const identity =`. :)

Reply:
done!